client/webserver/site/src/js/markets.ts

```diff
@@ -913,44 +952,36 @@ export default class MarketsPage extends BasePage {
   }
 
   /**
-   * previewQuoteAmt shows quote amount when rate or quantity input are changed
+   * previewTotal calculates Total value (in quote asset) for the order.
    */
-  previewQuoteAmt (show: boolean) {
+  previewTotal () {
     const page = this.page
-    if (!this.market.base || !this.market.quote) return // Not a supported asset
+    const market = this.market
+
     const order = this.parseOrder()
-    const adjusted = this.adjustedRate()
-    page.orderErr.textContent = ''
-    if (adjusted) {
-      if (order.sell) this.preSell()
-      else this.preBuy()
-    }
-    this.depthLines.input = []
-    if (adjusted && this.isLimit()) {
-      this.depthLines.input = [{
-        rate: order.rate / this.market.rateConversionFactor,
-        color: order.sell ? this.depthChart.theme.sellLine : this.depthChart.theme.buyLine
-      }]
-    }
-    this.drawChartLines()
-    if (!show || !adjusted || !order.qty) {
-      page.orderPreview.textContent = ''
-      this.drawChartLines()
-      return
-    }
-    const quoteAsset = app().assets[order.quote]
+
+    const quoteAsset = app().assets[market.quote.id]
     const quoteQty = order.qty * order.rate / OrderUtil.RateEncodingFactor
-    const total = Doc.formatCoinValue(quoteQty, this.market.quoteUnitInfo)
+    const total = Doc.formatCoinValue(quoteQty, market.quoteUnitInfo)
 
-    page.orderPreview.textContent = intl.prep(intl.ID_ORDER_PREVIEW, { total, asset: quoteAsset.symbol.toUpperCase() })
-    if (this.isSell()) this.preSell()
-    else this.preBuy()
```

Comment:
With `this.previewTotal()` actually, since there is no need to re-calculate `Max` when quantity/lots field changes.

And similar to `rateFieldChanged` func (like I mentioned above), `previewQuoteAmt` was doing a lot of different things which I also split into smaller ones to combine as needed.